src/internal/pfsdb/common.go

```diff
@@ -99,3 +99,10 @@ func (i *pageIterator[T]) next(ctx context.Context, extCtx sqlx.ExtContext) (*T,
 	i.pageIdx++
 	return &t, i.revision, nil
 }
+
+func IsNotFoundError(err error) bool {
+	return errors.As(err, &ErrRepoNotFound{}) ||
```

Comment:
I just realized something interesting. Here the `errors.As` we are using is actually a custom version in the `internal/errors` package. This works a bit differently from the standard lib `errors.As` in that the standard lib version would result in a panic because the target is a pointer to `ErrRepoNotFound` and `ErrReponotFound` does not implement the error interface. However, our version of `errors.As` checks the value of the target error which is `*ErrRepoNotFound`, and checks whether the value implements the error interface.

Anyways, I'm just surprised by the difference in behaviour.